pkg/controllers/state/cluster.go

```diff
@@ -354,21 +353,13 @@ func (c *Cluster) DeleteDaemonSet(key types.NamespacedName) {
 // you will hit race conditions and data corruption
 
 func (c *Cluster) newStateFromMachine(machine *v1alpha5.Machine, oldNode *StateNode) *StateNode {
+	var n *StateNode
 	if oldNode == nil {
-		oldNode = NewNode()
-	}
-	n := &StateNode{
-		Node:              oldNode.Node,
-		Machine:           machine,
-		hostPortUsage:     oldNode.hostPortUsage,
-		volumeUsage:       oldNode.volumeUsage,
-		daemonSetRequests: oldNode.daemonSetRequests,
-		daemonSetLimits:   oldNode.daemonSetLimits,
-		podRequests:       oldNode.podRequests,
-		podLimits:         oldNode.podLimits,
-		markedForDeletion: oldNode.markedForDeletion,
-		nominatedUntil:    oldNode.nominatedUntil,
+		n = NewNode()
+	} else {
+		n = oldNode.DeepCopy()
```

Comment:
This adds an extra copy of multiple maps that didn't used to be there.  The old code took advantage of the fact that the old node was not going to be used any longer and borrowed its internals.  Its cleaner to represent in language with move semantics.